addons/website/models/website_form.py

```diff
@@ -147,6 +149,26 @@ def init(self):
         self._cr.execute('ALTER TABLE ir_model_fields '
                          ' ALTER COLUMN website_form_blacklisted SET DEFAULT true')
 
+    @api.ondelete(at_uninstall=False)
+    def _check_if_used_in_website_form(self):
+        """prevent field deletion if used in a website form."""
+        form_views = self.env['ir.ui.view'].search([
+            ('type', '=', 'qweb'),
+            ('arch_db', 'like', 'data-model_name')
+        ])
```

Comment:
```suggestion
       arch_models = expression.OR([
           ('arch_db', 'like', f' data-model_name="{model}"')
        ] for model in self.mapped('model'))
        form_views = self.env['ir.ui.view'].search(expression.OR([
            [('type', '=', 'qweb')], arch_models
         ]))
```
Reduce lost of memory and time for looping all form views